### PR TITLE
Add top bar UI

### PR DESCRIPTION
**Pull request by kamirr**
Users can specify UI that will be added to the top bar of a node via the new `top_bar_ui` method in `NodeDataTrait`. A default definition is provided because it's not core to the functionality and so as to avoid unnecessarily breaking existing programs.

### CODE REVIEW

**Review by setzer22 — APPROVED** [review]
Looks good! I'd been thinking of adding something like this myself :)

**Comment by katopz** [comment]
Adding some example would be nice. 🤔

**Comment by kamirr** [comment]
@katopz I'm trying to figure out the best way to provide examples of the features from my recent PRs. I don't want to overload the main example by dumping even more stuff into it, as it's already a little daunting when starting out with `egui_node_graph`. On the other hand only adding code snippets to docs is limiting and the user won't have a runnable, tweakable example to fiddle with. tl;dr I'll make a separate PR for that.

**Comment by setzer22** [comment]
I concur. The example is meant to be used as a way to get started with the library, so I have avoided using some of the more complex features there. I'm not against adding more examples or docs (on the contrary!) but there should be at least one minimal example that people can use to get started.

>  tl;dr I'll make a separate PR for that.

Yup, sounds good. Let's keep PRs focused :+1:
